### DISCUSSION TITLE
Checksum for Telemetry packets, CFE, TLM and CCDS Spec Update

### DISCUSSION THREAD

**Discussion by shuq0** [nasa/cFS#483] [Ideas]
I am looking for some sort of checksum for the telemetry packet as with the command packets. The current thinking is to add a byte to the TLM secondary header and reuse the 8-bit checksum that is being used in the cmd checksum. This does break the CCDS spec and would require the spec to be updated to support this.

**Comment by skliper** [comment]
The CCSDS Space Packet Protocol (https://public.ccsds.org/Pubs/133x0b2e1.pdf) specifies the packet layout, see figure 4-1.  The "Packet Data Field", Packet Secondary Header and/or User Data Field are both variable sized and user defined. See also 4.1.4.2.3 which states the Ancillary Data Field (part of the secondary header) is not defined in the Recommended Standard.  So custom secondary headers are still CCSDS Space Packet Protocol compliant as long as they follow the other requirements in the standard.

**Reply by shuq0** [reply]
Thanks for the clarification, I was using CCDS V1 so I missed the additional field.

**Reply by skliper** [reply]
Note both V1 and V2 in cFS use secondary headers... just the primary CCSDS header is explicitly defined in the standard.  As in:
https://github.com/nasa/cFE/blob/81416f8f3552c2b0f85e820f4f423f62f8aee634/modules/msg/fsw/inc/ccsds_hdr.h#L51-L70
